src/AIAPI/MainGame/GameExtensions.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+// using ActionGame.Chara;
+using AIProject;
+using AIProject.SaveData;
+using AIChara;
+
+namespace KKAPI.MainGame
+{
+    /// <summary>
+    /// Extensions useful in the main game
+    /// </summary>
+    public static class GameExtensions
+    {
+
+        /// <summary>
+        /// Get the persisting heroine object that describes this character.
+        /// Returns null if the heroine could not be found. Works only in the main game.
+        /// </summary>
+        public static Actor GetHeroine(this ChaControl chaControl)
```

Comment:
Here's where the uncertinty comes into play.  Is the Actor class really the replacement for SaveData.Heroine in KK?

Reply:
AgentActor is more or less equal to NPC in KK, AgentActor.ActorData is more or less equal to Heroine. Look at it from the angle of a plugin wanting to do x to the character, which class does it need to access in these games?